### DISCUSSION TITLE
iasl error while dsdt patching asus z87-gryphon - Haswell

### DISCUSSION THREAD

**Discussion by KurdtNervanna** [xCuri0/ReBarUEFI#105] [Q&A]
After much retracing steps and trying to figure this error out I've hit a wall and was wondering if anyone else knows how to resolve this:

`PS C:\users\nucci\desktop\biosmod\iasl-win-20230628> ./iasl DSDT.aml

Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20230628
Copyright (c) 2000 - 2023 Intel Corporation

File appears to be binary: found 19096 non-ASCII characters, disassembling
Binary file appears to be a valid ACPI table, disassembling
Input file DSDT.aml, Length 0xE1EF (57839) bytes
ACPI: DSDT 0x0000000000000000 00E1EF (v02 ALASKA A M I    00000031 INTL 20091112)
Pass 1 parse of [DSDT]
Pass 2 parse of [DSDT]
Parsing Deferred Opcodes (Methods/Buffers/Packages/Regions)

Parsing completed

Found 12 external control methods, reparsing with new information
Pass 1 parse of [DSDT]
Pass 2 parse of [DSDT]
Parsing Deferred Opcodes (Methods/Buffers/Packages/Regions)

Parsing completed
Disassembly completed
ASL Output:    DSDT.dsl - 504301 bytes

 iASL Warning: There were 12 external control methods found during
 disassembly, but only 0 were resolved (12 unresolved). Additional
 ACPI tables may be required to properly disassemble the code. This
 resulting disassembler output file may not compile because the
 disassembler did not know how many arguments to assign to the
 unresolved methods. Note: SSDTs can be dynamically loaded at
 runtime and may or may not be available via the host OS.

 To specify the tables needed to resolve external control method
 references, the -e option can be used to specify the filenames.
 Example iASL invocations:
     iasl -e ssdt1.aml ssdt2.aml ssdt3.aml -d dsdt.aml
     iasl -e dsdt.aml ssdt2.aml -d ssdt1.aml
     iasl -e ssdt*.aml -d dsdt.aml

 In addition, the -fe option can be used to specify a file containing
 control method external declarations with the associated method
 argument counts. Each line of the file must be of the form:
     External (<method pathname>, MethodObj, <argument count>)
 Invocation:
     iasl -fe refs.txt -d dsdt.aml`

Specifically the "12 external errors" portion of the output in windows powershell...When recompiling this causes errors no matter what I do to fix them. The added DSDT lines of code aren't causing the error, the code in the dsdt.aml file itself. also if it matter or not this bios has an NVME .ffs added to it to allow nvme booting through a pcie slot.

Has anyone else run into this/know how to fix?

EDIT:

For sake of completeness for the post, here is the compile error i get after if i try to compile an edited/unedited .aml file:

`Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20230628
Copyright (c) 2000 - 2023 Intel Corporation

Error    6126 -  Compiler aborting due to parser-detected syntax error(s)

DSDTMod.dsl  10439:         }
Error    6126 -            ^ syntax error, unexpected '}'

DSDTMod.dsl  14487:
Error    6126 - syntax error, unexpected $end and premature End-Of-File

Error    6126 -  Compiler aborting due to parser-detected syntax error(s)

DSDTMod.dsl  10439:         }
Error    6126 -            ^ syntax error, unexpected '}'

DSDTMod.dsl  14487:
Error    6126 - syntax error, unexpected $end and premature End-Of-File

Input file:    DSDTMod.dsl - Compilation aborted due to parser-detected syntax error(s)

Compilation failed. 3 Errors, 0 Warnings, 0 Remarks
No AML files were generated due to syntax error(s)`

**Comment by xCuri0** [comment]
It says how to fix all these errors at the bottom of the DSDT page
